editor/components/DarkModeToggle.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import Form from "react-bootstrap/Form";
+import Dropdown from "react-bootstrap/Dropdown";
+import { FaMoon } from "react-icons/fa6";
+
+export const darkModelocalStorageKey = "userDarkModeSetting";
+
+/**
+ * Dropdown item with a switch that enables dark mode
+ */
+export default function DarkModeToggle() {
+  const [appColorMode, setAppColorMode] = useState<"dark" | "light">(
+    localStorage.getItem(darkModelocalStorageKey) === "dark" ? "dark" : "light"
+  );
+
+  useEffect(() => {
+    const htmlElement = document.documentElement;
+    htmlElement.setAttribute("data-bs-theme", appColorMode);
+    localStorage.setItem(darkModelocalStorageKey, appColorMode);
+  }, [appColorMode]);
+
+  return (
+    <Dropdown.Item
+      className="d-flex align-items-center"
+      onClick={(e) => {
+        setAppColorMode(appColorMode === "dark" ? "light" : "dark");
+        e.preventDefault();
+        e.stopPropagation();
+      }}
+    >
+      <FaMoon className="me-3" />
+      <span className="me-3">Dark mode</span>
+      <Form.Check
```

Comment:
small thing, but I see the warning below. The `readOnly` prop that it seems to fix. BTW i like that the entire row is clickable. ✨

<img width="327" alt="Screenshot 2025-02-28 at 1 01 58 PM" src="https://github.com/user-attachments/assets/26d2f50d-3e35-4e41-92c3-4011cba73730" />